packages/apollo-server-core/src/runHttpQuery.ts

```diff
@@ -20,6 +20,11 @@ export interface HttpQueryRequest {
   request: Pick<Request, 'url' | 'method' | 'headers'>;
 }
 
+export interface HttpQueryResponse {
+  gqlResponse: string;
+  responseInit: ResponseInit;
```

Comment:
The typing here is a little funky, since ResponseInit.headers is a HeadersInit which allows various options (Headers object, array of arrays, dictionary object) but the consumers assume that it's the latter. I'm almost surprised this typechecks, though maybe TS lets you `[]` index anything?  Not sure what to do about this — maybe just roll your own type, or at least comment on it you can assume that the headers field is a record?

Reply:
Makes sense, we'll roll our own for now